hw/chisel/src/main/scala/snax/streamer/Streamer.scala

```diff
@@ -0,0 +1,429 @@
+    var flattenedIndex = -1
+    val flattenedSeq = inputSeq.zipWithIndex.flatMap { case (size, dimIndex) =>
+      (0 until size).map { innerIndex =>
+        flattenedIndex = flattenedIndex + 1
+        (dimIndex, innerIndex, flattenedIndex)
+      }
+    }
+
+    flattenedSeq
+  }
+
+  // data reader <> TCDM read ports
+  val read_flatten_seq = flattenSeq(params.dataReaderTcdmPorts)
+  for ((dimIndex, innerIndex, flattenedIndex) <- read_flatten_seq) {
+    // read request to TCDM
+    io.data.tcdm_req(flattenedIndex) <> data_reader(dimIndex).io.tcdm_req(
+      innerIndex
+    )
+
+    // signals from TCDM responses
+    data_reader(dimIndex).io.tcdm_rsp(innerIndex) <> io.data.tcdm_rsp(
+      flattenedIndex
+    )
+  }
+
+  // data writer <> TCDM write ports
+  // TCDM request port bias based on the read TCDM ports number
+  val write_flatten_seq = flattenSeq(params.dataWriterTcdmPorts)
+  for ((dimIndex, innerIndex, flattenedIndex) <- write_flatten_seq) {
+    // write request to TCDM
+    io.data.tcdm_req(flattenedIndex + tcdm_read_ports_num) <> data_writer(
+      dimIndex
+    ).io.tcdm_req(innerIndex)
+  }
+
+}
+
+// Scala main function for generating test streamer system verilog file
+object StreamerTester extends App {
+  emitVerilog(
+    new Streamer(StreamerParams()),
+    Array("--target-dir", "generated/streamer/tester")
+  )
+}
+
+// Scala main function for generating system verilog file for different accelerators
+// including GEMM, Post-processing SIMD and MAC engine
+object GemmStreamer extends App {
+  emitVerilog(
+    new Streamer(
+      StreamerParams(
+        temporalAddrGenUnitParams =
+          GeMMStreamerParameters.temporalAddrGenUnitParams,
+        fifoReaderParams = GeMMStreamerParameters.fifoReaderParams,
+        fifoWriterParams = GeMMStreamerParameters.fifoWriterParams,
+        stationarity = GeMMStreamerParameters.stationarity,
+        dataReaderParams = GeMMStreamerParameters.dataReaderParams,
+        dataWriterParams = GeMMStreamerParameters.dataWriterParams
+      )
+    ),
+    Array("--target-dir", "generated/streamer/gemm")
+  )
+}
+
+object PostProcessingStreamer extends App {
+  emitVerilog(
+    new Streamer(
+      StreamerParams(
+        temporalAddrGenUnitParams =
+          PostProcessingStreamerParameters.temporalAddrGenUnitParams,
+        fifoReaderParams = PostProcessingStreamerParameters.fifoReaderParams,
+        fifoWriterParams = PostProcessingStreamerParameters.fifoWriterParams,
+        stationarity = PostProcessingStreamerParameters.stationarity,
+        dataReaderParams = PostProcessingStreamerParameters.dataReaderParams,
+        dataWriterParams = PostProcessingStreamerParameters.dataWriterParams
+      )
+    ),
+    Array("--target-dir", "generated/streamer/pp")
+  )
+}
+
+object MacStreamer extends App {
+  emitVerilog(
+    new Streamer(
+      StreamerParams()
+    ),
+    Array("--target-dir", "generated/streamer/mac")
+  )
+}
```

Comment:
should these be moved to their relative owners?
`StreamerTester` -> StreamerTest.scala
`GemmStreamer` -> Gemm.scala (?)
...

Or is this for some reason not possible?

Reply:
Are we not moving everything into the same repository for this reason? So will this be possible now?